src/plugins/api/qmlpluginapi.cpp

```diff
@@ -118,14 +120,41 @@ QQmlListProperty<Score> PluginAPI::scores()
 bool PluginAPI::writeScore(Score* s, const QString& name, const QString& ext)
 {
     if (!s || !s->score()) {
+        LOGW("PluginAPI::writeScore: no score provided");
         return false;
     }
 
-    UNUSED(name);
-    UNUSED(ext);
+    if (s->score() != currentScore()) {
+        LOGW("PluginAPI::writeScore: only writing the selected score is currently supported");
+        return false;
+    }
 
-    NOT_IMPLEMENTED;
-    return false;
+    const notation::INotationPtr notation = context()->currentNotation();
+
+    if (!notation) {
+        LOGW("PluginAPI::writeScore: no notation found");
+        return false;
+    }
+
+    const project::INotationWriterPtr writer = writers()->writer(ext.toStdString());
+
+    if (!writer) {
+        LOGW("PluginAPI::writeScore: failed to create writer for the format '%s'", ext.toUtf8().constData());
+        return false;
+    }
+
+    project::INotationWriter::UnitType unitType;
```

Comment:
I think, for better readability, we can move the resolving of unitType into a separate method

Reply:
OK, I moved this code to a separate method.